src/Mod/TechDraw/Gui/Command.cpp

```diff
@@ -801,6 +805,34 @@ bool _checkDrawViewPartBalloon(Gui::Command* cmd) {
     return true;
 }
 
+bool _checkDirectPlacement(const QGIViewPart *viewPart, const std::vector<std::string> &subNames, QPointF &placement)
```

Comment:
Please add some comments that others looking into the code maybe years later quickly understand what this code is for. 

For example for edges you set the midpoint as anchor. There is no option to change this and I think there is a good reason. This is for example worth comments.
Also what about when I select an area, this case is not handled, this is also worth a comment...